app/assets/javascripts/float-kit/addon/modifiers/apply-floating-ui.js

```diff
@@ -0,0 +1,109 @@
+import Modifier from "ember-modifier";
+import { registerDestructor } from "@ember/destroyable";
+import { bind } from "discourse-common/utils/decorators";
+import {
+  arrow,
+  autoPlacement,
+  autoUpdate,
+  computePosition,
+  inline,
+  offset,
+  shift,
+} from "@floating-ui/dom";
+import { headerOffset } from "discourse/lib/offset-calculator";
+import { iconHTML } from "discourse-common/lib/icon-library";
+import { cached } from "@glimmer/tracking";
+import domFromString from "discourse-common/lib/dom-from-string";
+
+export default class FloatKitApplyFloatingUi extends Modifier {
+  constructor(owner, args) {
+    super(owner, args);
+    registerDestructor(this, (instance) => instance.cleanup());
+  }
+
+  modify(element, [trigger, options]) {
+    this.element = element;
+    this.trigger = trigger;
+    this.options = options ?? {};
+
+    if (this.options.arrow) {
+      element.appendChild(this.arrowElement);
+    }
+
+    this.cleanup = autoUpdate(trigger, element, this.update);
+  }
+
+  @cached
+  get arrowElement() {
+    return domFromString(
+      iconHTML("tippy-rounded-arrow", { class: "arrow" })
+    )[0];
+  }
+
+  @bind
+  update() {
```

Comment:
would love to see this `async'd`

Reply:
you mean more like integrated with runloop? because async by itself is not going to do much

Reply:
oh yes indeed, I actually forgot computePosition was a promise :+1: